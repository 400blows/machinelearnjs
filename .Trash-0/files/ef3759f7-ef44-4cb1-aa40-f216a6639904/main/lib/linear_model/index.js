"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const base_1 = require("./base");
const lr = new base_1.LinearRegression();
const X = [0.5, 1, 1.5, 2, 2.5];
const y = [0, 1, 2, 3, 4];
lr.fit({ X, y });
console.log('checking pred X:', lr.predict(3));
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvbGliL2xpbmVhcl9tb2RlbC9pbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOztBQUFBLGlDQUEwQztBQUUxQyxNQUFNLEVBQUUsR0FBRyxJQUFJLHVCQUFnQixFQUFFLENBQUM7QUFFbEMsTUFBTSxDQUFDLEdBQUcsQ0FBQyxHQUFHLEVBQUUsQ0FBQyxFQUFFLEdBQUcsRUFBRSxDQUFDLEVBQUUsR0FBRyxDQUFDLENBQUM7QUFDaEMsTUFBTSxDQUFDLEdBQUcsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxDQUFDLENBQUM7QUFFMUIsRUFBRSxDQUFDLEdBQUcsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxDQUFDO0FBQ2pCLE9BQU8sQ0FBQyxHQUFHLENBQUMsa0JBQWtCLEVBQUUsRUFBRSxDQUFDLE9BQU8sQ0FBQyxDQUFDLENBQUMsQ0FBQyxDQUFDIn0=