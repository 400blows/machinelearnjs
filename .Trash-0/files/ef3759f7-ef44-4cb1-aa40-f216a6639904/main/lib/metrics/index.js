"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const ConfusionMatrix = __importStar(require("ml-confusion-matrix"));
/**
 * Reference: http://www.dataschool.io/simple-guide-to-confusion-matrix-terminology/
 * Official API Doc: https://mljs.github.io/confusion-matrix/
 * @param yTrue
 * @param yPred
 */
exports.confusion_matrix = ConfusionMatrix;
// const trueLabels =      [0, 1, 0, 1, 1, 0];
const trueLabels = [[0, 1, 1], [1, 0, 0]];
const predictedLabels = [1, 1, 1, 1, 0, 0];
const cm2 = new exports.confusion_matrix([[13, 2], [10, 5]], ['cheese', 'dog']);
console.log('acc', cm2.getAccuracy());
console.log(cm2.getFalseCount());
console.log(cm2.getIndex('dog'));
const classification_1 = require("./classification");
const accResult = classification_1.accuracyScore({
    y_true: [0, 1, 2, 3],
    y_pred: [0, 2, 1, 3]
});
console.log('accuracy result ', accResult);
const accResultNorm = classification_1.accuracyScore({
    y_true: [0, 1, 2, 3],
    y_pred: [0, 2, 1, 3],
    normalize: false
});
console.log('accuracy result iwht norm false ', accResultNorm);
const loss_zero_one_result = classification_1.zeroOneLoss({
    y_true: [1, 2, 3, 4],
    y_pred: [2, 2, 3, 5]
});
console.log('loss zero one ', loss_zero_one_result);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvbGliL21ldHJpY3MvaW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7Ozs7Ozs7O0FBQUEscUVBQXVEO0FBRXZEOzs7OztHQUtHO0FBQ1UsUUFBQSxnQkFBZ0IsR0FBRyxlQUFlLENBQUM7QUFFaEQsOENBQThDO0FBQzlDLE1BQU0sVUFBVSxHQUFHLENBQUMsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLENBQUMsQ0FBQyxDQUFDO0FBQzFDLE1BQU0sZUFBZSxHQUFHLENBQUMsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLENBQUMsQ0FBQztBQUUzQyxNQUFNLEdBQUcsR0FBRyxJQUFJLHdCQUFnQixDQUFDLENBQUMsQ0FBQyxFQUFFLEVBQUUsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFLEVBQUUsQ0FBQyxDQUFDLENBQUMsRUFBRSxDQUFDLFFBQVEsRUFBRSxLQUFLLENBQUMsQ0FBQyxDQUFDO0FBRXhFLE9BQU8sQ0FBQyxHQUFHLENBQUMsS0FBSyxFQUFFLEdBQUcsQ0FBQyxXQUFXLEVBQUUsQ0FBQyxDQUFDO0FBQ3RDLE9BQU8sQ0FBQyxHQUFHLENBQUMsR0FBRyxDQUFDLGFBQWEsRUFBRSxDQUFDLENBQUM7QUFDakMsT0FBTyxDQUFDLEdBQUcsQ0FBQyxHQUFHLENBQUMsUUFBUSxDQUFDLEtBQUssQ0FBQyxDQUFDLENBQUM7QUFFakMscURBQThEO0FBRTlELE1BQU0sU0FBUyxHQUFHLDhCQUFhLENBQUM7SUFDOUIsTUFBTSxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxDQUFDO0lBQ3BCLE1BQU0sRUFBRSxDQUFDLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsQ0FBQztDQUNyQixDQUFDLENBQUM7QUFFSCxPQUFPLENBQUMsR0FBRyxDQUFDLGtCQUFrQixFQUFFLFNBQVMsQ0FBQyxDQUFDO0FBRTNDLE1BQU0sYUFBYSxHQUFHLDhCQUFhLENBQUM7SUFDbEMsTUFBTSxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxDQUFDO0lBQ3BCLE1BQU0sRUFBRSxDQUFDLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsQ0FBQztJQUNwQixTQUFTLEVBQUUsS0FBSztDQUNqQixDQUFDLENBQUM7QUFFSCxPQUFPLENBQUMsR0FBRyxDQUFDLGtDQUFrQyxFQUFFLGFBQWEsQ0FBQyxDQUFDO0FBRS9ELE1BQU0sb0JBQW9CLEdBQUcsNEJBQVcsQ0FBQztJQUN2QyxNQUFNLEVBQUUsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLENBQUM7SUFDcEIsTUFBTSxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxDQUFDO0NBQ3JCLENBQUMsQ0FBQztBQUVILE9BQU8sQ0FBQyxHQUFHLENBQUMsZ0JBQWdCLEVBQUUsb0JBQW9CLENBQUMsQ0FBQyJ9