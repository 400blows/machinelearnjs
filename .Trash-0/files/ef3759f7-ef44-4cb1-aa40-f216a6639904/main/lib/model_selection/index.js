"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const _split_1 = require("./_split");
const X = [[0, 1], [2, 3], [4, 5], [6, 7], [8, 9]];
const y = [0, 1, 2, 3, 4];
console.log('original X', X);
console.log('original y', y);
console.log(_split_1.train_test_split(X, y, {
    test_size: 0.33,
    train_size: 0.67,
    random_state: 42
}));
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvbGliL21vZGVsX3NlbGVjdGlvbi9pbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOztBQUFBLHFDQUE0QztBQUU1QyxNQUFNLENBQUMsR0FBRyxDQUFDLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxDQUFDLENBQUM7QUFDbkQsTUFBTSxDQUFDLEdBQUcsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxDQUFDLENBQUM7QUFFMUIsT0FBTyxDQUFDLEdBQUcsQ0FBQyxZQUFZLEVBQUUsQ0FBQyxDQUFDLENBQUM7QUFDN0IsT0FBTyxDQUFDLEdBQUcsQ0FBQyxZQUFZLEVBQUUsQ0FBQyxDQUFDLENBQUM7QUFDN0IsT0FBTyxDQUFDLEdBQUcsQ0FDVCx5QkFBZ0IsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFO0lBQ3JCLFNBQVMsRUFBRSxJQUFJO0lBQ2YsVUFBVSxFQUFFLElBQUk7SUFDaEIsWUFBWSxFQUFFLEVBQUU7Q0FDakIsQ0FBQyxDQUNILENBQUMifQ==