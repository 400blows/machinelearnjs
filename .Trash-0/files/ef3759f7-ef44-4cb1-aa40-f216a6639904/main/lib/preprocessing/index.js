"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const data_1 = require("./data");
// Playing around with onehotencoder
console.log('Playing with one hot encoder');
const enc = new data_1.OneHotEncoder();
const planetList = [
    { planet: 'mars', isGasGiant: false, value: 10 },
    { planet: 'saturn', isGasGiant: true, value: 20 },
    { planet: 'jupiter', isGasGiant: true, value: 30 }
];
const encodeInfo = enc.encode(planetList, {
    dataKeys: ['value', 'isGasGiant'],
    labelKeys: ['planet']
});
console.log(encodeInfo.data);
const decodedInfo = enc.decode(encodeInfo.data, encodeInfo.decoders);
console.log(decodedInfo);
// MinMaxScaler
const minmaxScaler = new data_1.MinMaxScaler({ featureRange: [0, 1] });
minmaxScaler.fit([4, 5, 6]);
const result = minmaxScaler.fit_transform([4, 5, 6]);
console.log(result);
// Imputer
const Imputer_1 = require("./Imputer");
const testX = [[1, 2], [null, 3], [7, 6]];
const imp = new Imputer_1.Imputer({ missingValues: null, axis: 0 });
imp.fit(testX);
const impResult = imp.fit_transform([[null, 2], [6, null], [7, 6]]);
console.log('checking result', impResult);
// Binarizer
const data_2 = require("./data");
const binX = [[1, -1, 2], [2, 0, 0], [0, 1, -1]];
const newBin = new data_2.Binarizer({ threshold: 0 });
const binResult = newBin.transform(binX);
console.log('binresult: ', binResult);
// Label Encoder
const label_1 = require("./label");
const le = new label_1.LabelEncoder();
const labelX = ['amsterdam', 'paris', 'tokyo'];
le.fit(labelX);
const transformX = ['tokyo', 'tokyo', 'paris'];
const leResult = le.transform(transformX);
console.log(leResult);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvbGliL3ByZXByb2Nlc3NpbmcvaW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFBQSxpQ0FBcUQ7QUFFckQsb0NBQW9DO0FBQ3BDLE9BQU8sQ0FBQyxHQUFHLENBQUMsOEJBQThCLENBQUMsQ0FBQztBQUM1QyxNQUFNLEdBQUcsR0FBRyxJQUFJLG9CQUFhLEVBQUUsQ0FBQztBQUNoQyxNQUFNLFVBQVUsR0FBRztJQUNqQixFQUFFLE1BQU0sRUFBRSxNQUFNLEVBQUUsVUFBVSxFQUFFLEtBQUssRUFBRSxLQUFLLEVBQUUsRUFBRSxFQUFFO0lBQ2hELEVBQUUsTUFBTSxFQUFFLFFBQVEsRUFBRSxVQUFVLEVBQUUsSUFBSSxFQUFFLEtBQUssRUFBRSxFQUFFLEVBQUU7SUFDakQsRUFBRSxNQUFNLEVBQUUsU0FBUyxFQUFFLFVBQVUsRUFBRSxJQUFJLEVBQUUsS0FBSyxFQUFFLEVBQUUsRUFBRTtDQUNuRCxDQUFDO0FBQ0YsTUFBTSxVQUFVLEdBQUcsR0FBRyxDQUFDLE1BQU0sQ0FBQyxVQUFVLEVBQUU7SUFDeEMsUUFBUSxFQUFFLENBQUMsT0FBTyxFQUFFLFlBQVksQ0FBQztJQUNqQyxTQUFTLEVBQUUsQ0FBQyxRQUFRLENBQUM7Q0FDdEIsQ0FBQyxDQUFDO0FBQ0gsT0FBTyxDQUFDLEdBQUcsQ0FBQyxVQUFVLENBQUMsSUFBSSxDQUFDLENBQUM7QUFFN0IsTUFBTSxXQUFXLEdBQUcsR0FBRyxDQUFDLE1BQU0sQ0FBQyxVQUFVLENBQUMsSUFBSSxFQUFFLFVBQVUsQ0FBQyxRQUFRLENBQUMsQ0FBQztBQUNyRSxPQUFPLENBQUMsR0FBRyxDQUFDLFdBQVcsQ0FBQyxDQUFDO0FBRXpCLGVBQWU7QUFDZixNQUFNLFlBQVksR0FBRyxJQUFJLG1CQUFZLENBQUMsRUFBRSxZQUFZLEVBQUUsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxDQUFDO0FBQ2hFLFlBQVksQ0FBQyxHQUFHLENBQUMsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsQ0FBQyxDQUFDLENBQUM7QUFDNUIsTUFBTSxNQUFNLEdBQUcsWUFBWSxDQUFDLGFBQWEsQ0FBQyxDQUFDLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxDQUFDLENBQUMsQ0FBQztBQUNyRCxPQUFPLENBQUMsR0FBRyxDQUFDLE1BQU0sQ0FBQyxDQUFDO0FBRXBCLFVBQVU7QUFDVix1Q0FBb0M7QUFFcEMsTUFBTSxLQUFLLEdBQUcsQ0FBQyxDQUFDLENBQUMsRUFBRSxDQUFDLENBQUMsRUFBRSxDQUFDLElBQUksRUFBRSxDQUFDLENBQUMsRUFBRSxDQUFDLENBQUMsRUFBRSxDQUFDLENBQUMsQ0FBQyxDQUFDO0FBQzFDLE1BQU0sR0FBRyxHQUFHLElBQUksaUJBQU8sQ0FBQyxFQUFFLGFBQWEsRUFBRSxJQUFJLEVBQUUsSUFBSSxFQUFFLENBQUMsRUFBRSxDQUFDLENBQUM7QUFDMUQsR0FBRyxDQUFDLEdBQUcsQ0FBQyxLQUFLLENBQUMsQ0FBQztBQUNmLE1BQU0sU0FBUyxHQUFHLEdBQUcsQ0FBQyxhQUFhLENBQUMsQ0FBQyxDQUFDLElBQUksRUFBRSxDQUFDLENBQUMsRUFBRSxDQUFDLENBQUMsRUFBRSxJQUFJLENBQUMsRUFBRSxDQUFDLENBQUMsRUFBRSxDQUFDLENBQUMsQ0FBQyxDQUFDLENBQUM7QUFDcEUsT0FBTyxDQUFDLEdBQUcsQ0FBQyxpQkFBaUIsRUFBRSxTQUFTLENBQUMsQ0FBQztBQUUxQyxZQUFZO0FBQ1osaUNBQW1DO0FBRW5DLE1BQU0sSUFBSSxHQUFHLENBQUMsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLENBQUMsQ0FBQyxDQUFDLENBQUM7QUFDakQsTUFBTSxNQUFNLEdBQUcsSUFBSSxnQkFBUyxDQUFDLEVBQUUsU0FBUyxFQUFFLENBQUMsRUFBRSxDQUFDLENBQUM7QUFDL0MsTUFBTSxTQUFTLEdBQUcsTUFBTSxDQUFDLFNBQVMsQ0FBQyxJQUFJLENBQUMsQ0FBQztBQUN6QyxPQUFPLENBQUMsR0FBRyxDQUFDLGFBQWEsRUFBRSxTQUFTLENBQUMsQ0FBQztBQUV0QyxnQkFBZ0I7QUFDaEIsbUNBQXVDO0FBRXZDLE1BQU0sRUFBRSxHQUFHLElBQUksb0JBQVksRUFBRSxDQUFDO0FBQzlCLE1BQU0sTUFBTSxHQUFHLENBQUMsV0FBVyxFQUFFLE9BQU8sRUFBRSxPQUFPLENBQUMsQ0FBQztBQUMvQyxFQUFFLENBQUMsR0FBRyxDQUFDLE1BQU0sQ0FBQyxDQUFDO0FBQ2YsTUFBTSxVQUFVLEdBQUcsQ0FBQyxPQUFPLEVBQUUsT0FBTyxFQUFFLE9BQU8sQ0FBQyxDQUFDO0FBQy9DLE1BQU0sUUFBUSxHQUFHLEVBQUUsQ0FBQyxTQUFTLENBQUMsVUFBVSxDQUFDLENBQUM7QUFDMUMsT0FBTyxDQUFDLEdBQUcsQ0FBQyxRQUFRLENBQUMsQ0FBQyJ9