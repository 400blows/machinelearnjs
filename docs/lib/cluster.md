[kalimdor](../README.md) > ["cluster/k_means"](../modules/_lib_cluster_k_means_.md)

# cluster/k_means

## Index

### Variables

* [data](_lib_cluster_k_means_.md#data)
* [vectors](_lib_cluster_k_means_.md#vectors)

---

## Variables

<a id="data"></a>

### `<Const>` data

**● data**: *`object`[]* =  [
  { company: 'Microsoft', size: 91259, revenue: 60420 },
  { company: 'IBM', size: 400000, revenue: 98787 },
  { company: 'Skype', size: 700, revenue: 716 },
  { company: 'SAP', size: 48000, revenue: 11567 },
  { company: 'Yahoo!', size: 14000, revenue: 6426 },
  { company: 'eBay', size: 15000, revenue: 8700 }
]

*Defined in [lib/cluster/k_means.ts:5](https://github.com/JasonShin/kalimdorjs/blob/12bcdb8/src/lib/cluster/k_means.ts#L5)*

___
<a id="vectors"></a>

### `<Const>` vectors

**● vectors**: *`any`[]* =  []

*Defined in [lib/cluster/k_means.ts:15](https://github.com/JasonShin/kalimdorjs/blob/12bcdb8/src/lib/cluster/k_means.ts#L15)*

___

