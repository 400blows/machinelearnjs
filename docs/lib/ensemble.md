[kalimdor](../README.md) > [ensemble/RandomForestClassifier](../classes/_lib_ensemble_forest_.randomforestclassifier.md)

# Class: RandomForestClassifier

## Hierarchy

**RandomForestClassifier**

## Index

### Constructors

* [constructor](_lib_ensemble_forest_.randomforestclassifier.md#constructor)

### Methods

* [fit](_lib_ensemble_forest_.randomforestclassifier.md#fit)
* [predict](_lib_ensemble_forest_.randomforestclassifier.md#predict)

---

## Constructors

<a id="constructor"></a>

###  constructor

⊕ **new RandomForestClassifier**(props?: *`object`*): [RandomForestClassifier](_lib_ensemble_forest_.randomforestclassifier.md)

*Defined in [lib/ensemble/forest.ts:6](https://github.com/JasonShin/kalimdorjs/blob/12bcdb8/src/lib/ensemble/forest.ts#L6)*

RandomForestClassifier hey bro

**Parameters:**

| Param | Type | Default value |
| ------ | ------ | ------ |
| `Default value` props | `object` |  { nEstimator: 10 } |

**Returns:** [RandomForestClassifier](_lib_ensemble_forest_.randomforestclassifier.md)

___

## Methods

<a id="fit"></a>

###  fit

▸ **fit**(__namedParameters: *`object`*): `void`

*Defined in [lib/ensemble/forest.ts:21](https://github.com/JasonShin/kalimdorjs/blob/12bcdb8/src/lib/ensemble/forest.ts#L21)*

Build a forest of trees from the training set (X, y).

**Parameters:**

| Param | Type |
| ------ | ------ |
| __namedParameters | `object` |

**Returns:** `void`
void

___
<a id="predict"></a>

###  predict

▸ **predict**(X: *`any`*): `any`[]

*Defined in [lib/ensemble/forest.ts:44](https://github.com/JasonShin/kalimdorjs/blob/12bcdb8/src/lib/ensemble/forest.ts#L44)*

Predict class for X.

The predicted class of an input sample is a vote by the trees in the forest, weighted by their probability estimates. That is, the predicted class is the one with highest mean probability estimate across the trees.

**Parameters:**

| Param | Type | Description |
| ------ | ------ | ------ |
| X | `any` |  array-like or sparse matrix of shape = \[n_samples\] |

**Returns:** `any`[]

___

