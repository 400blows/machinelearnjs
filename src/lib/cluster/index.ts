import { installTF } from '../utils/deps';
installTF();
import { KMeans } from './k_means';

export { KMeans };
