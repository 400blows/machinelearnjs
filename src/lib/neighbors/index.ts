import { installTF } from '../utils/deps';
installTF();
import { KNeighborsClassifier } from './classification';

export { KNeighborsClassifier };
