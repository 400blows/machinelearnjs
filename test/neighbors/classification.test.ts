import * as _ from 'lodash';
import { KNeighborsClassifier } from '../../src/lib/neighbors';
import { matchExceptionWithSnapshot } from '../util_testing';

describe('classification:KNeighborsClassifier', () => {
  const X1 = [[0, 0, 0], [0, 1, 1], [1, 1, 0], [2, 2, 2], [1, 2, 2], [2, 1, 2]];
  const y1 = [0, 0, 0, 1, 1, 1];

  const X2 = [['a', 'b', 'c'], ['b', 'c', 'd']];
  const y2 = ['a', 'z'];
  it('should predict 1 for [1, 2] against the sample 1', () => {
    const knn = new KNeighborsClassifier();
    knn.fit(X1, y1);
    const expected = 1;
    const pred = knn.predict([1, 2]);
    expect(pred).toBe(expected);
  });

  it('should predict 0 for [0, 0, 0] against the sample 1', () => {
    const knn = new KNeighborsClassifier();
    knn.fit(X1, y1);
    const pred = knn.predict([0, 0, 0]);
    const expected = 0;
    expect(pred).toBe(expected);
  });

  it('should predict [ 1, 0, 1 ] for [ [1, 2], [0, 2], [9, 5] ]  against the sample 1', () => {
    const knn = new KNeighborsClassifier();
    knn.fit(X1, y1);
    const pred = knn.predict([[1, 2], [0, 2], [9, 5]]);
    const expected = [1, 0, 1];
    expect(_.isEqual(pred, expected)).toBe(true);
  });

  it('should fit X2 and predict y2', () => {
    const knn = new KNeighborsClassifier();
    knn.fit(X2, y2);
    const pred = knn.predict([['a', 'b', 'c']]);
    const expected = ['a'];
    expect(pred).toEqual(expected);
  });

  it('should predict [ 1, 0, 1 ] for [ [1, 2, 4], [0], [9, 5] ] against the sample 1', () => {
    const knn = new KNeighborsClassifier();
    expect(() => knn.predict([[1, 2, 4], [0], [9, 5]])).toThrow(
      'Element arr[1] should have 3 elements, but has 1 elements'
    );
  });

  it('should throw an error if X is not a matrix', () => {
    const knn = new KNeighborsClassifier();
    matchExceptionWithSnapshot(knn.fit, [[1], [2]]);
  });

  it('should throw an error if y is not a vector', () => {
    const knn = new KNeighborsClassifier();
    matchExceptionWithSnapshot(knn.fit, [[[1], [2]], 123]);
  });
});
