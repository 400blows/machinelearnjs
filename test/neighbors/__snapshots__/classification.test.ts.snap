// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classification:KNeighborsClassifier should throw an error if X is not a matrix 1`] = `[Error: The matrix is not 2D shaped: [1] of [1]]`;

exports[`classification:KNeighborsClassifier should throw an error if y is not a vector 1`] = `[Error: validateMatrix1D has received a non-array argument]`;
